maro/simulator/scenarios/supply_chain/units/distribution.py

```diff
@@ -36,15 +45,18 @@ def __init__(
             id, data_model_name, data_model_index, facility, parent, world, config,
         )
 
-        # Vehicle unit dict of this distribution unit. Key: vehicle type; Value: a list of vehicle instances.
-        self.vehicles: Dict[str, List[VehicleUnit]] = defaultdict(list)
+        self._vehicle_num: Dict[str, Union[int, float]] = {}
```

Comment:
Vehicle num could be a float?

Reply:
It could be math.inf. And inf is a float, not int

Reply:
I personally prefer `Optional[int]` and use `None` to represent infinity (so we need special judge when using it). Float here is a little bit confusing.